packages/salesforcedx-vscode-soql/src/queryResultsView/queryDataViewService.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { JsonMap } from '@salesforce/ts-types';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { getDocumentName } from '../commonUtils';
+import {
+  DATA_VIEW_MEDIA_PATH,
+  QUERY_DATA_VIEW_PANEL_TITLE,
+  QUERY_DATA_VIEW_SCRIPT_FILENAME,
+  QUERY_DATA_VIEW_STYLE_FILENAME,
+  QUERY_DATA_VIEW_TYPE,
+  TABULATOR_SCRIPT_FILENAME,
+  TABULATOR_STYLE_FILENAME
+} from '../constants';
+import { HtmlUtils } from '../editor/htmlUtils';
+import { html } from './queryDataHtml';
+
+export class QueryDataViewService {
+  public currentPanel: vscode.WebviewPanel | undefined = undefined;
+  public readonly viewType = QUERY_DATA_VIEW_TYPE;
+  public static extensionPath: string;
+
+  constructor(
+    private subscriptions: vscode.Disposable[],
+    private queryData: JsonMap[],
+    private document: vscode.TextDocument
+  ) {}
+
+  public static register(context: vscode.ExtensionContext): void {
+    QueryDataViewService.extensionPath = context.extensionPath;
+  }
+
+  private updateWebviewWith(webview: vscode.Webview, queryData: JsonMap[]) {
+    webview.postMessage({
+      type: 'update',
+      data: queryData,
+      documentName: getDocumentName(this.document)
+    });
+  }
+
+  public createOrShowWebView(): vscode.Webview {
+    this.currentPanel = vscode.window.createWebviewPanel(
+      this.viewType,
+      QUERY_DATA_VIEW_PANEL_TITLE,
+      vscode.ViewColumn.Two,
```

Comment:
Would this open in the expected column when soql file is already in a split editor?

Reply:
It opens in a new tab if the editor is already split. We plan to create a WI to handle the webview visibility and moving better.
![image](https://user-images.githubusercontent.com/25188632/94072648-51b4d400-fdb3-11ea-9eda-6eb0b4b25fc5.png)